doc/contributing.rst

```diff
@@ -21,6 +21,30 @@ of the users who use the package.
 page as the maintainers about changes or enhancements before too much
 coding is done saves everyone time and effort!
 
+Installing developer version
+----------------------------
+
+The standard installation procedure can be found on the
+[MNE website](https://martinos.org/mne/stable/install_mne_python.html)
+but if you want to install a developer version directly from the source
+code in order to contribute, you can follow those steps:
+
+  .. code-block:: console
+
+     $ git clone git@github.com:mne-tools/mne-python.git
+     $ cd mne-python
+     $ conda env create -f dev_environment.yml
+     $ conda activate mne-dev-py36
+     $ pip install -e .
```

Comment:
rather than duplicating the env file that will be easily get out of sync I would suggest to use the standard env and start by uninstalling mne stable before installing dev version.

Reply:
or another option is to keep it like this but to add a test to make sure that the diff from environment.yml and dev_environment.yml is just 2 lines (change env title + remove mne)

thoughts?

Reply:
There had to be more stuff than 2 lines (all the extra dependencies to compile the doc etc) but I'm not sure that the dev_environment.yml proposed by @GuillaumeFavelier had them all. I did not test.

See https://github.com/mne-tools/mne-python/issues/5767#issuecomment-443793098 That's what I usually do. I'm not sure if there is any difference between conda-forge mne and environment.yml

But we do more stuff in our CIs than just remove mne.